lib/Sema/TypeChecker.h

```diff
@@ -536,6 +536,9 @@ enum TypeResolutionFlags : unsigned {
 
   /// Whether we are checking the parameter list of a subscript.
   TR_SubscriptParameters = 0x2000000,
+
+  /// Is it okay to resolve an IUO sigil ("!") here?
+  TR_AllowIUO = 0x4000000,
```

Comment:
I'm assuming plumbing this through to be checked in more places and ripping out the old diagnostic is still pending?

Reply:
Yes. I'll probably look at that next although I have a couple other changes I could finish up and push as well.